resolver-dns/src/main/java/io/netty/resolver/dns/ThreadLocalNameServerAddressStream.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+import io.netty.util.concurrent.FastThreadLocal;
+
+import java.net.InetSocketAddress;
+
+/**
+ * A thread local based address stream for a specific hostname.
+ */
+class ThreadLocalNameServerAddressStream implements DnsServerAddressStream {
+
+    private final String hostname;
+    private final DnsServerAddressStreamProvider dnsServerAddressStreamProvider;
+    private final FastThreadLocal<DnsServerAddressStream> threadLocal = new FastThreadLocal<DnsServerAddressStream>() {
```

Comment:
we should be scared of the always-increasing sequence on fast-thread local while used as an instance field here?

Reply:
Also TBH I am not convinced that using a FastThreadLocal here is actually relevant since resolver query does not seem to be used on the hot path. Perhaps Netty should go away from this usage, @trustin introduced this change a few years ago can give his opinion.

Reply:
Need to dig into this later this week... Maybe we can get rid of it all together

Reply:
@normanmaurer query usage remains useful we are actually using it in Vert.x DNS client tho